rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,505 @@
+- Make sure hidden popuops or titles are accessible through keyboard. 
+
+# Focus management
+
+> Specification
+
+**Target:** Everyone and especially people with visual impairments or cognitive limitations, motor disabilities.  
+**Goal:** Setting guidelines for general focus behaviours & Providing screen reader & keyboard users a smooth experience by managing focus behavior for certain elements like Modals/Page transition etc. 
+
+Well-planned focus management is important to ensuring a comfortable user experience & to guide the user through the intended flow of the app.
+
+Focus management goes hand in hand with keyboard accessibility, so there will be interoperability between the two. 
+
+### Areas to cover
+
+- [Focus order](#keyboard-tab-order)
+- [Focus ring styling](#focus-ring-styling)
+- [Focus traps](#focus-trap)
+  - Focus restoration
+- Page transitions
+
+## Focus order
+
+Focus order or tab order we already covered in [keyboard tab order section](#keyboard-tab-order)
+
+## Focus ring styling
+
+The focus ring must be visible to all users, 
+By default, the browser uses the user agent specific focus styling. But this behaviour can be overriden with CSS. 
+
+Best practices for focus styling:
+
+- Provide focus styles that are highly visible.
+- Make sure that a visible element has focus at all times when using a keyboard.
+- Avoid using `*:focus { outline: none }` snippet to hide focus
+- Design focus rings such that it has proper contrast
+
+> Not providing visible focus ring is [Failure of Success Criterion 2.4.7](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78)
+
+**Keyboard only focus rings**
+
+While focus indicators are neccesary,  
+Mouse clicks also changes the focus, for example clicking on a button also makes the focus ring visible.  
+
+This behavior can be undesirable from design perspective, preferring focus styling to only be present only if a keyboard is used can be a solution. 
+
+<img width="45%" src="./images/accessibility/focus-visible-demo.gif" alt="Demo of focus visible css pesudo class, showing two buttons and comparing the results" />
+
+**Implementations**
+
+- CSS
+- ReactAria
+
+**CSS:**  
+
+> [Demo](https://codesandbox.io/s/blade-a11y-css-focus-ring-051xff?file=/src/App.tsx)
+
+CSS provides us with a pesudo class called [`:focus-visible`](https://css-tricks.com/keyboard-only-focus-styles). Browser support for focus-visible is [good enough.](https://caniuse.com/?search=%3Afocus-visible).  
+And we can also use the [official pollyfill](https://github.com/WICG/focus-visible) for older browsers.
+
+**ReactAria:**  
+
+> [Demo](https://codesandbox.io/s/blade-a11y-reactaria-focus-ring-hc3cub?file=/src/App.tsx)
+
+ReactAria provides a [FocusRing](https://react-spectrum.adobe.com/react-aria/FocusRing.html) component which solves this. 
+
+While this works great, this solution can be a bit overkill since react aria does this all with javascript and have it's own event handling system.  
```

Comment:
```suggestion
While this works great, this solution can be a bit of an overkill since ReactAria does this all with JavaScript and has its own event handling system.  
```